File: VB/Controllers/HomeController.vb
Imports System.Web.Script.Serialization

Public Class HomeController
    Inherits System.Web.Mvc.Controller

    Public Function Index() As ActionResult
        'default data
        Dim model As OrdersModel = New OrdersModel With { _
            .StartDate = New DateTime(1996, 8, 1), _
            .EndDate = New DateTime(1996, 8, 30) _
        }

        Return View(model)
    End Function

    <HttpPost()> _
    Public Function GridViewPartial() As ActionResult
        ' http://msdn.microsoft.com/en-us/library/bb299886.aspx 
        Dim serializer As New JavaScriptSerializer()

        Dim model As OrdersModel = New OrdersModel With { _
            .StartDate = serializer.Deserialize(Of DateTime)(Request.Params("StartDate")).ToLocalTime(), _
            .EndDate = serializer.Deserialize(Of DateTime)(Request.Params("EndDate")).ToLocalTime() _
        }

        Return PartialView("_GridView", model.Data)
    End Function
End Class

File: CS/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using Example.Models;
using System.Web.Script.Serialization;
using System.Globalization;

namespace Example.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            //default data
            OrdersModel model = new OrdersModel {
                StartDate = new DateTime(1996, 8, 1),
                EndDate = new DateTime(1996, 8, 30)
            };

            return View(model);
        }

        [HttpPost]
        public ActionResult GridViewPartial() {
            OrdersModel model = new OrdersModel {
                StartDate = DateTime.ParseExact(Request.Params["StartDate"], "d|M|yyyy", CultureInfo.InvariantCulture),
                EndDate = DateTime.ParseExact(Request.Params["EndDate"], "d|M|yyyy", CultureInfo.InvariantCulture)
            };
            
            return PartialView("_GridView", model.Data);
        }
    }
}
